strax/io.py

```diff
@@ -14,20 +14,33 @@
 
 blosc.set_releasegil(True)
 
+BASE_CONDITIONS = [
+    lambda x: isinstance(x, np.ndarray),
+    lambda x: len(x)>0,
+]
+BLOSC_CONDITIONS = BASE_CONDITIONS + [lambda x: x.nbytes<2**31]
 
 COMPRESSORS = dict(
     bz2=dict(
         compress=bz2.compress,
-        decompress=bz2.decompress),
+        decompress=bz2.decompress,
+        conditions=BASE_CONDITIONS),
     zstd=dict(
         compress=zstd.compress,
-        decompress=zstd.decompress),
+        decompress=zstd.decompress,
+        conditions=BASE_CONDITIONS),
     blosc=dict(
         compress=partial(blosc.compress, shuffle=False),
-        decompress=blosc.decompress),
-    lz4=dict(compress=lz4.compress, decompress=lz4.decompress)
+        decompress=blosc.decompress,
+        conditions=BLOSC_CONDITIONS),
+    lz4=dict(compress=lz4.compress,
+     decompress=lz4.decompress, 
+     conditions=BASE_CONDITIONS)
```

Comment:
Missing some whitespace here, looks like. Different tab sizes? Use spaces instead anyway, python complains.

Reply:
thats super weird, spaces shouldnt matter inside a dictionary definition. what error did you get? or are you talking about a linter error?